.github/workflows/myfirst.yml

```diff
@@ -1,12 +1,46 @@
-name: Continuous Integration
-on: [push]
-
+name: CI
+on: 
+  push:
+    paths-ignore:
+    - '**/README.md'
 jobs:
-  build:
-    name: Build and test
+  build-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker build --target test --tag my-test-image .
+      - run: docker run my-test-image
+
+  prod-check:
+    #if: ${{ github.ref == 'refs/heads/main' }}
+    if: ${{ github.ref == 'refs/heads/exercise-9' }}
     runs-on: ubuntu-latest
+    needs: build-test
     steps:
-    - uses: actions/checkout@v2
-    #- run: echo Hello World
-    - run: docker build --target test --tag my-test-image .
-    - run: docker run my-test-image
\ No newline at end of file
+      - run: echo "Deploying to production server on branch $GITHUB_REF"
+
+  build-push:
+    runs-on: ubuntu-latest
+    needs: prod-check
+    
+    steps:
+      - uses: actions/checkout@v3
+      #- uses: actions/checkout@5f6796be8905cf35cb55bea0a4e62158c4bf116d 
+      #- uses: actions/checkout@exercise-9
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
```

Comment:
(Discussed off the review, this action doesn't seem to be working, but we can add appropriate arguments using `cache-from` as you'd identified to achieve the aims)